src/ui/src/wds/WdsIcons/LucideIcon.vue

```diff
@@ -0,0 +1,48 @@
+<!-- eslint-disable vue/no-v-html -->
+
+<template>
+	<span :style="{ display: 'contents' }" v-html="rawHtml" />
+</template>
+
+<script setup lang="ts">
+import { nextTick, onMounted, ref, useAttrs, watch } from "vue";
+import { createIcons, icons } from "lucide";
+
+defineOptions({
+	inheritAttrs: false,
+});
+
+const props = defineProps({
+	name: { type: String, required: true },
+});
+
+const attrs = useAttrs();
+
+const rawHtml = ref("");
+
+function renderIcon(name: string) {
+	rawHtml.value = `<i data-lucide="${name}"></i>`;
+
+	nextTick(() => {
+		createIcons({
+			icons,
+			attrs: {
+				width: "1em",
+				height: "1em",
+				...attrs,
+			},
+		});
+	});
+}
+
+watch(
+	() => props.name,
+	(newName) => {
+		renderIcon(newName);
+	},
+);
+
+onMounted(() => {
+	renderIcon(props.name);
+});
```

Comment:
You can use `immediate` instead of mixing `watch` / `onMounted`. Also,It can be shortened.

```suggestion
watch(toRef(props, 'name'),  renderIcon, { immediate: true });
```

Reply:
Great catch, I will apply it!

Originally the logic relied on the DOM element being available, which is why I used onMounted. But since that changed, I didn't update the implementation.

Reply:
We might also need to add `flush: 'post'` to ensure the DOM is ready before calling `createIcons`, since it relies on actual elements being in place - https://vuejs.org/guide/essentials/watchers.html#post-watchers